util/openssl-format-check.pl

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/perl
+#
+# Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+# openssl-format-check.pl
+# - check source file(s) according to OpenSSL coding style
+#
+# usage:
+#   openssl-format-check.pl <files>
+#
+# checks adherence to the formatting rules of the OpenSSL coding guidelines.
+# This tool is in preliminary state: it is incomplete and yields some false
+# positives. Still it should be useful for detecting most typical glitches.
+
+use strict;
+use List::Util qw[min max];
+
+use constant INDENT_LEVEL => 4;
+use constant MAX_LENGTH => 80;
+
+my $indent;
+my $line;
+my $line_opening_brace;
+my $contents_before;       # used only if $line > 1
+my $contents_before2;      # used only if $line > 2
+my $hanging_indent;
+my $hanging_parens;        # used only if $hanging_indent != -1
+my $hanging_braces;        # used only if $hanging_indent != -1
+my $hanging_alt_indent;    # used only if $hanging_indent != -1
+my $braceless_indent;
+my $multiline_condition_indent;
+my $in_multiline_macro;    # number of lines so far within macro
+my $in_multiline_comment;
+my $comment_indent;        # used only if $in_multiline_comment == 1
+
+sub reset_file_state {
+    $indent = 0;
+    $line = 0;
+    $line_opening_brace = 0;
+    $hanging_indent = -1;
+    $braceless_indent = 0;
+    $multiline_condition_indent = -1;
+    $in_multiline_macro = 0;
+    $in_multiline_comment = 0;
+}
+
+reset_file_state();
+while(<>) {
+    $line++;
+    my $orig_ = $_;
+
+    if(m/[\x09]/) {
```

Comment:
A block comment like "# Look for illegal characters" before clumps of lines would be helpful.  (I prefer that to inline right-margin comments but YMMV)

Especially starting around 132 etc.

Reply:
I've meanwhile added block comments and improved some comments and variable names.